src/components/PopupWithForm.js

```diff
@@ -21,12 +24,19 @@ export default class PopupWithForm extends Popup {
     this._popupForm.addEventListener("submit", (event) => {
       event.preventDefault();
       this._handleFormSubmit(this._getInputValues());
-      this.close();
     });
   }
 
   close() {
     this._popupForm.reset();
     super.close();
   }
+
+  setLoading(isLoading) {
+    if (isLoading) {
+      this._saveButton.textContent = "Saving...";
```

Comment:
**Could be improved**

`"Saving..."` is a hardcode here. You can change it as well


```js

  setLoading(isLoading, loadingText = "Saving...") {
    if (isLoading) {
      this._saveButton.textContent = loadingText;
    } else {
      this._saveButton.textContent = this._saveButtonText;
    }
  }

```